### ISSUE TITLE
receive-message not working

### ISSUE THREAD

**Issue by vikkio88** [softwaremill/elasticmq#129]
Hi,
I don't know if this is a problem with elasticmq or with the Docker image that is wrapping it that I am using, but I can send-message without any problems, both with awscli and a php script wrapping the sdk, but I cannot receive them, they always come back as an empty set
```
vikkio@dell [16:16:09] [~] 
-> % aws --endpoint-url http://localhost:9324 sqs send-message --queue-url http://localhost:9324/queue/default --message-body "a message 2"
{
    "MD5OfMessageBody": "970d370372f570f25ff372c1bfde3b74",
    "MD5OfMessageAttributes": "d41d8cd98f00b204e9800998ecf8427e",
    "MessageId": "26b1cdfd-b26a-496e-9078-23c20e76a607"
}
vikkio@dell [16:16:14] [~]  
-> % aws --endpoint-url http://localhost:9324 sqs receive-message --queue-url http://localhost:9324/queue/default         
```
Has this happened to anyone else recently?

**Comment by ankon** [comment]
Some quick questions:
1. Do you see anything in the output of elasticmq? If so, please post it :)
2. Are you by chance having another consumer on the same queue that is consuming the messages before you get to see them?

**Comment by vikkio88** [comment]
Hi @ankon 
1. it is wrapped inside https://github.com/roribio/alpine-sqs so I can't see any output/logs
2. no, for sure nothing else is consuming those messages, I thought that too, as this docker image has an interface to see the messages on the queues, bu it is happening as well with https://github.com/vsouza/docker-SQS-local that has for sure nothing reading from the queues and it is using elasticmq too.

**Comment by ankon** [comment]
> it is wrapped inside https://github.com/roribio/alpine-sqs so I can't see any output/logs

You can: Either run without detaching the container -- ` docker run -p 9324:9324 vsouza/sqs-local`, or use `docker logs CONTAINER_ID` with the id of the sqs-local container (check `docker ps`).

**Comment by vikkio88** [comment]
I tried to get the logs, I think the error is in the https://github.com/roribio/alpine-sqs docker cointainer.

Using docker-SQS-local I can receive the message but I get something like
```
Error receiving message from queue Attribute MD5 mismatch. Expected d41d8cd98f00b204e9800998ecf8427e, found No AttributesError receiving message from queue Attribute MD5 mismatch. Expected d41d8cd98f00b204e9800998ecf8427e, found No AttributesError receiving message from queue Attribute MD5 mismatch. Expected d41d8cd98f00b204e9800998ecf8427e, found No AttributesError receiving message from queue Attribute MD5 mismatch. Expected d41d8cd98f00b204e9800998ecf8427e, found No AttributesError receiving message from queue Attribute MD5 mismatch. Expected d41d8cd98f00b204e9800998ecf8427e, found No Attributes
```
Which looks to me more like an application error

**Comment by crutis** [comment]
This was happening to me when I used `.withMessageAttributeNames("All")` in my request. Switching to specific header names worked for me.

**Comment by agwhitaker93** [comment]
It looks like this issue is caused by `org.elasticmq.rest.sqs.MD5Util#md5AttributeDigest`
`md5.update` on line 308 seems to be getting null

```
➜ md5sum /dev/null
d41d8cd98f00b204e9800998ecf8427e  /dev/null
```

As a short term fix, providing message attributes sidesteps the issue

**Comment by vikkio88** [comment]
@agwhitaker93 I dont think they care about this issue tbh, on another note, you work for SykesCottages, say "ciao" to Phil from me lol

**Comment by adamw** [comment]
@vikkio88 We care about all issues across all of our projects, but time's limited. Help is always welcome! PRs always get priority :)

**Comment by agwhitaker93** [comment]
I've raised a PR that fixes the issue as best as I can tell: https://github.com/softwaremill/elasticmq/pull/289

**Comment by adamw** [comment]
Released in 0.14.14, thanks!
